core/src/main/java/org/apache/iceberg/util/DateTimeUtil.java

```diff
@@ -88,4 +88,8 @@ public static long microsFromTimestamptz(OffsetDateTime dateTime) {
   public static String formatTimestampMillis(long millis) {
     return DATE_FORMAT.format(LocalDateTime.ofInstant(Instant.ofEpochMilli(millis), ZoneOffset.UTC));
   }
+
+  public static String formatTimestampMillisWithLocalTime(long millis) {
+    return DATE_FORMAT.format(LocalDateTime.ofInstant(Instant.ofEpochMilli(millis), ZoneOffset.systemDefault()));
```

Comment:
Iceberg does not produce zone-specific time formats. This should not go in a common class like `DateTimeUtil`.